_posts/2020/11/2020-11-06-configure-artipie-maven-repo.md

```diff
@@ -0,0 +1,124 @@
+author: genryxy
+description: This blog post explains how to use Artipie as a Maven repository.
+keywords:
+  - artipie
+  - private
+  - maven
+  - repository
+  - registry
+  - java
+---
+
+
+As we know, Apache Maven is one of the most popular build management tool for Java. Many companies that do Java development have private Maven repositories, where they store all the artifacts. There are some problems with management when many developers work with a huge repository.    
+Let's assume, you need to restrict access to artifacts. So, it's necessary to split this one into several small repositories or add authentication for users. But at the same time you want to keep the ability to share easily and quickly within a team or between multiple teams of developers. Or you are just wondering how to set up an own repository for a pet project.  
+[Artipie](https://github.com/artipie/artipie) allows you to solve these problems as it is a binary artifact management tool. In the blog post I will share with you how to configure Artipie and project for this purpose. Let's start!
+
+In the simplest case, you need a file with the contents below. We want to use file storage therefore we specify `fs` as the type. At first, create a separate directory with the following `artipie.yaml` file inside:
+
+```yml
+meta:
+  storage:
+    type: fs
+    path: /var/artipie/configs
+```
+
+After that you can install and run Artipie with [Docker](https://docs.docker.com/get-started/):
+
+```bash
+$ docker run -d --name artipie \
+             -v $(pwd)/artipie.yaml:/etc/artipie.yml \
+             -v $(pwd):/var/artipie \
+             -p 8080:80 \
+             artipie/artipie:latest
+```
+
+In the previous step, we created a file with settings for Artipie. Now let's see what the configuration file for the repository should look like. For this, we need to create `configs` directory and the following `my-maven.yaml` file inside:
+
+```yml
+repo:
+  type: maven
+  storage:
+    type: fs
+    path: /var/artipie/data
+```
+As a results, you will have the following files structure:
+
+```bash
+.
+├── artipie.yaml
+├── configs
+    └── my-maven.yaml
+```
+
+At this step, I will show you how to configure your project to use created Artipie Maven repository. If you want to `mvn deploy` your project to Artipie, add the following to project's `pom.xml`:
+
+```xml
+<distributionManagement>
+    <repository>
+        <id>artipie</id>
+        <url>http://localhost:8080/my-maven</url>
+    </repository>
+</distributionManagement>
+```
+
+In case you want to `mvn install` deployed packages from Artipie then add the following to project's `pom.xml`:
+
+```xml
+<repositories>
+    <repository>
+        <id>artipie</id>
+        <url>http://localhost:8080/my-maven</url>
+    </repository>
+</repositories>
+```
+If you have completed all the instructions correctly now you should be able to use the repository: deploy and install Maven artifacts.
+
+**Configure Artipie with authentication**
```

Comment:
@genryxy this time it's better to make it a header, I think. Using BOLD text instead of native Markdown formatting is a bad practice